rxmongo/src/main/scala/akka/contrib/persistence/mongodb/RxMongoJournaller.scala

```diff
@@ -47,7 +49,9 @@ class RxMongoJournaller(driver: RxMongoDriver) extends MongoPersistenceJournalli
   private[mongodb] override def batchAppend(writes: ISeq[AtomicWrite])(implicit ec: ExecutionContext):Future[ISeq[Try[Unit]]] = {
     val batch = writes.toStream.map(aw => Try(driver.serializeJournal(Atom[BSONDocument](aw, driver.useLegacySerialization))))
     Future.sequence(batch.map {
-      case Success(document:BSONDocument) => journal.insert(document, writeConcern).map(writeResultToUnit)
```

Comment:
Oh and to be clear - I agree the ordering is correct.  We should write to journal before realtime.